### PR TITLE
[RDY] Add an about dialog. 

### PR DESCRIPTION
**Pull request by ohuang12**
Following items added:
- Help menu --> currently only having the about option.
- File menu --> exit option.

### CODE REVIEW

**Comment by pvanagtmaal** [comment]
Note that the PR involves more than just an about dialog.

File: src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	public AboutDialog() {
+		setupDialog();
+	}
+
+	private void setupDialog() {
+		dialog = new Dialog<String>();
+		dialog.setTitle("About DNAinator");
+		dialog.getDialogPane().getStylesheets().add(getClass().getResource(STYLE).toString());
+		Stage stage = (Stage) dialog.getDialogPane().getScene().getWindow();
+		Image icon = new Image(this.getClass().getResourceAsStream(ICON));
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java]: 2 lines up you don't use `this`, here you do. Try to be consistent in your style.

File: src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.stage.Stage;
+
+/**
+ * Creates a {@link Dialog} displaying information about the
+ * application.
+ */
+public class AboutDialog {
+	private Dialog<String> dialog;
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java]: For our other dialogs, we use the newer [Alert](http://download.java.net/jdk9/jfxdocs/javafx/scene/control/Alert.html). You might consider using this as well

File: src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+	private void setupDialog() {
+		dialog = new Dialog<String>();
+		dialog.setTitle("About DNAinator");
+		dialog.getDialogPane().getStylesheets().add(getClass().getResource(STYLE).toString());
+		Stage stage = (Stage) dialog.getDialogPane().getScene().getWindow();
+		Image icon = new Image(this.getClass().getResourceAsStream(ICON));
+		stage.getIcons().add(icon);
+		
+		ImageView img = new ImageView();
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java]: the image created at [line 39](https://github.com/DNAinator/dnainator/pull/54/files#diff-946d2d1cf8f9be155f9e6c3c6890143cR39) can be passed to the constructor.

File: src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		StringBuilder contents = new StringBuilder();
+		contents.append("Version: " + prop.getProperty("version") + "\n");
+		contents.append("Jente Hidskes, Gerlof Fokkema, Owen Huang, "
+				+ "Skip Lentz, Piet van Agtmaal");
+		return contents.toString();
+	}
+	
+	private void readProperties() {
+		prop = new Properties();
+		InputStream inputStream = getClass().getClassLoader().getResourceAsStream(PROPERTIES);
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java]: This stream can probably be created using [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html).

File: src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+		if (inputStream != null) {
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java]: Why wouldn't you want to catch the null as well? IMO, it also should show an "Error reading properties".

File: src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+			try {
+				prop.load(inputStream);
+			} catch (IOException e) {
+				new ExceptionDialog(e, "Error reading properties");
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java]: Missing a "." at the end of the error message.

File: src/main/java/nl/tudelft/dnainator/ui/widgets/ExceptionDialog.java [hunk 7]
```diff
@@ -37,6 +41,10 @@ public ExceptionDialog(Throwable throwable, String title) {
 		content.add(textArea, 0, 1);
 
 		alert.getDialogPane().setExpandableContent(content);
+		Stage stage = (Stage) alert.getDialogPane().getScene().getWindow();
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/ExceptionDialog.java]: What is this for? If you want to set an icon for the Stage, please do so in the `DNAInator` class.

File: src/main/java/nl/tudelft/dnainator/ui/widgets/ProgressDialog.java [hunk 8]
```diff
@@ -33,8 +36,11 @@ private void setupAlert() {
 		alert.setTitle(" ");
 		alert.setHeaderText("Loading...");
 		alert.getButtonTypes().add(ButtonType.CANCEL);
-
+		
 		alert.getDialogPane().setContent(progressBar);
+		Stage stage = (Stage) alert.getDialogPane().getScene().getWindow();
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/ProgressDialog.java]: What is this for? If you want to set an icon for the Stage, please do so in the `DNAInator` class.

File: src/main/resources/ui/style.css [hunk 9]
```diff
@@ -1,11 +1,22 @@
+.dialog-pane .button-bar {
+    -fx-border-color: #D4D4D4 #FFFFFF #FFFFFF #FFFFFF;
```
Comment (pvanagtmaal) [inline src/main/resources/ui/style.css]: It seems like you are mixing tab/space indentation in this file.

File: src/main/java/nl/tudelft/dnainator/ui/widgets/ProgressDialog.java [hunk 10]
```diff
@@ -30,10 +33,11 @@ public ProgressDialog(Service service) {
 
 	private void setupAlert() {
 		alert = new Alert(Alert.AlertType.NONE);
-		alert.setTitle(" ");
+		alert.setTitle("DNAinator");
 		alert.setHeaderText("Loading...");
 		alert.getButtonTypes().add(ButtonType.CANCEL);
-
+		System.out.println(parent.getScene() == null);
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/ProgressDialog.java]: Missed this after debugging? ;)

File: src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java [hunk 11]
```diff
@@ -33,13 +35,12 @@
 	 * Constructs a WindowController object, creating
 	 * a {@link FileLoadService} to go with it.
 	 */
-	@FXML
```
Comment (Unia) [inline src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java]: I'm not sure that this is a good solution...

File: src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+import javafx.scene.control.ButtonBar.ButtonData;
+import javafx.scene.control.ButtonType;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+
+/**
+ * Creates a {@link Alert} displaying information about the
+ * application.
+ */
+public class AboutDialog {
```
Comment (Unia) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java]: Can't we just extend `Alert`? (Should also apply this to `ExceptionDialog` then)

File: src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+			new ExceptionDialog(parent, e, "Error reading properties.");
+		}
+	}
+
+	/**
+	 * Shows the {@link Alert} if it is not null.
+	 * The {@link Alert} will block until user input is received. 
+	 */
+	public void show() {
+		if (alert != null) {
```
Comment (Unia) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/AboutDialog.java]: Can it ever be null? If so, shouldn't we recreate it in that case?

**Comment by Unia** [comment]
Looks good to me now! :+1:

File: src/main/resources/ui/style.css [hunk 14]
```diff
@@ -1,11 +1,22 @@
...
+	-fx-font-size: 38px;
 	-fx-font-weight: bold;
-	-fx-text-fill: #333333;
-	-fx-alignment: center;
-	-fx-label-padding: 10 10 10 10;
 }
 
+.test .label {
+	-fx-alignment: top-left;
+}
```
Comment (Balletie) [inline src/main/resources/ui/style.css]: Is this still supposed to be here?

**Pull request merged**